build/Microsoft.DotNet.Cli.Compile.targets

```diff
@@ -0,0 +1,184 @@
+                                           SetupStage;
+                                           CompileStage;
+                                           BuildProjectsForNuGetPackages;
+                                           GenerateNuGetPackagesArchive">
+  </Target>
+
+  <Target Name="SetupStage">
+    <PropertyGroup>
+      <SrcDirectory>$(RepoRoot)/src</SrcDirectory>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <CompileStageInputs Include="$(SrcDirectory)/**/*.cs" Exclude="$(SrcDirectory)/**/obj/**/*.cs" />
+      <CompileStageInputs Include="$(SrcDirectory)/**/project.json;$(SrcDirectory)/**/project.lock.json" />
+
+      <Stage Include="Stage1">
+        <Inputs>@(CompileStageInputs)</Inputs>
+        <StageDirectory>$(Stage1Directory)</StageDirectory>
+        <StageSymbolsDirectory>$(Stage1SymbolsDirectory)</StageSymbolsDirectory>
+        <DotnetExe>$(Stage0Path)/dotnet.exe</DotnetExe>
+      </Stage>
+      <Stage Include="Stage2">
+        <Inputs>@(CompileStageInputs)</Inputs>
+        <StageDirectory>$(Stage2Directory)</StageDirectory>
+        <StageSymbolsDirectory>$(Stage2SymbolsDirectory)</StageSymbolsDirectory>
+        <DotnetExe>$(DotnetStage1)</DotnetExe>
+      </Stage>          
+    </ItemGroup>    
+  </Target>    
+
+  <Target Name="CompileStage"
+          Inputs="%(Stage.Inputs)"
+          Outputs="%(StageDirectory)/sdk/$(SdkVersion)/dotnet.dll">
+      <PropertyGroup>      
+        <ExeSuffix Condition=" '$(OSName)' == 'win' ">.exe</ExeSuffix>
+        <ExeSuffix Condition=" '$(OSName)' != 'win' "></ExeSuffix>
+
+        <DynamicLibPrefix Condition=" '$(OSName)' == 'win' "></DynamicLibPrefix>
+        <DynamicLibPrefix Condition=" '$(OSName)' != 'win' ">lib</DynamicLibPrefix>
+
+        <DynamicLibSuffix Condition=" '$(OSName)' == 'win' ">.dll</DynamicLibSuffix>
+        <DynamicLibSuffix Condition=" '$(OSName)' == 'osx' ">.dylib</DynamicLibSuffix>
+        <DynamicLibSuffix Condition=" '$(DynamicLibSuffix)' == '' ">.so</DynamicLibSuffix>
+
+        <DotnetHostBaseName>dotnet$(ExeSuffix)</DotnetHostBaseName>
+        <DotnetHostFxrBaseName>$(DynamicLibPrefix)hostfxr$(DynamicLibSuffix)</DotnetHostFxrBaseName>
+        <HostPolicyBaseName>$(DynamicLibPrefix)hostpolicy$(DynamicLibSuffix)</HostPolicyBaseName>
+
+        <StageDirectory>%(Stage.StageDirectory)</StageDirectory>
+        <StageSymbolsDirectory>%(Stage.StageSymbolsDirectory)</StageSymbolsDirectory>
+        <SdkOutputDirectory>$(StageDirectory)/sdk/$(SdkVersion)</SdkOutputDirectory>            
+        <BinaryToCorehostifyRelDir>runtimes/any/native</BinaryToCorehostifyRelDir>
+        <BinaryToCorehostifyOutDir>$(SdkOutputDirectory)/$(BinaryToCorehostifyRelDir)</BinaryToCorehostifyOutDir>
+        <MSBuildTargetsDirectory>$(SdkOutputDirectory)/runtimes/any/native</MSBuildTargetsDirectory>
+
+        <SharedFrameworkName>Microsoft.NETCore.App</SharedFrameworkName>
+        <SharedFrameworkNuGetVersion>1.0.0</SharedFrameworkNuGetVersion>
```

Comment:
We already have this in build.proj.  You can just use that instead of duplicating it here.